### PR TITLE
Enhance c/c++ layer

### PR DESCRIPTION
**Pull request by chenlijun99**
### PR Prelude

Thank you for working on SpaceVim! :)

Please complete these steps and check these boxes before filing your PR:

- [x] I have read and understood SpaceVim's [CONTRIBUTING](https://github.com/SpaceVim/SpaceVim/blob/master/CONTRIBUTING.md) document.
- [x] I have read and understood SpaceVim's [CODE_OF_CONDUCT](https://github.com/SpaceVim/SpaceVim/blob/master/CODE_OF_CONDUCT.md) document.
- [x] I understand my PR may be closed if it becomes obvious I didn't actually perform all of these steps.

### Why this change is necessary and useful?

[Please explain **in detail** why the changes in this PR are needed.]

### CODE REVIEW

File: autoload/SpaceVim/layers/lang/c.vim [hunk 1]
```diff
@@ -88,6 +88,10 @@ function! SpaceVim#layers#lang#c#plugins() abort
 endfunction
 
 function! SpaceVim#layers#lang#c#config() abort
+  call SpaceVim#mapping#gd#add('c',
+        \ function('s:go_to_def'))
```
Comment (wsdjeg) [inline autoload/SpaceVim/layers/lang/c.vim:92]: we should defind two function, here is a example, user who enable lsp for c but not for cpp. he should be able to use lsp gd

File: autoload/SpaceVim/layers/lang/c.vim [hunk 2]
```diff
@@ -194,3 +200,10 @@ function! s:update_neoinclude(argv, fts) abort
   let b:neoinclude_paths = path
 endfunction
 
+function! s:go_to_def() abort
+  if !SpaceVim#layers#lsp#check_filetype('c') && !SpaceVim#layers#lsp#check_filetype('cpp')
```
Comment (wsdjeg) [inline autoload/SpaceVim/layers/lang/c.vim]: If you do not want to use two function, you can change this line to:

```
if !SpaceVim#layers#lsp#check_filetype(&ft)
```

**Comment by chenlijun99** [comment]
Fixed

**Comment by wsdjeg** [comment]
as said in https://spacevim.org/development/#language-specified-key-bindings

we should unify the language specified key bindings for different language layer.

**Comment by chenlijun99** [comment]
Yeah, but it's not mentioned which key binding to use for "show references".
I used "SPC l \*", because vim's default behavior with "*" is "search for same word under the cursor in the buffer"

**Comment by wsdjeg** [comment]
`*` is hard to press, you need to press `Shift+8`. in typescript layer, we are using `SPC l r`, but I think `SPC l r` is not a good idea, it should be used to run file.

**Comment by chenlijun99** [comment]
Uh, I didn't take that into consideration. Yeah, Vim's philosophy is "stay on the home row".

What do you think about "SPC l f", where `f` stands for "find"?

**Comment by wsdjeg** [comment]
`SPC l f` is ok to me. :)

________________________________
wsdjeg@outlook.com

**Comment by chenlijun99** [comment]
Done

**Pull request merged**